### PR TITLE
Refactors for last child and left sibling pointer insertions

### PR DESCRIPTION
**Pull request by Raj-Datta-Manohar**
Added last child pointer, left sibling pointer in JsonShredder
Added some insert and remove tests

### CODE REVIEW

File: bundles/sirix-core/src/test/java/org/sirix/access/node/json/JsonNodeTrxRemoveTest.java [hunk 1]
```diff
@@ -254,6 +384,36 @@ private void assertsForRemoveObjectAsRightSibling(JsonNodeReadOnlyTrx rtx) {
...
+
+    rtx.moveTo(16);
+
+    assertEquals(4, rtx.getChildCount());
+    assertEquals(6, rtx.getDescendantCount());
+    assertEquals(20, rtx.getFirstChildKey());
+
+    rtx.moveTo(20);
+
+    assertEquals(-1, rtx.getLeftSiblingKey());
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/test/java/org/sirix/access/node/json/JsonNodeTrxRemoveTest.java:400]: You could also used Fixed.NULL_NODE_KEY.getStandardProperty() here :-)

File: bundles/sirix-core/src/test/resources/json/test.json [hunk 2]
```diff
@@ -1 +1 @@
-{"foo":["bar",null,2.33],"bar":{"hello":"world","helloo":true},"baz":"hello","tada":[{"foo":"bar"},{"baz":false},"boo",{},[]]}
\ No newline at end of file
+{"foo":["bar",null,2.33],"bar":{"hello":"world","helloo":true},"baz":"hello","tada":[{"foo":"bar"},{"baz":false},"boo",[],{}]}
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/test/resources/json/test.json:1]: Is this really the case, that the empty object and empty array changed the position?

Comment (Raj-Datta-Manohar) [reply bundles/sirix-core/src/test/resources/json/test.json:1]: Yes, I've changed it for remove last child as array test

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/test/resources/json/test.json:1]: Ah okay 👍

**Comment by JohannesLichtenberger** [comment]
Thanks :+1:

**Pull request merged**